sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -95,13 +101,37 @@ public Result implement(EnumerableRelImplementor implementor, Prefer prefer) {
       Thread.currentThread().setContextClassLoader(BeamEnumerableConverter.class.getClassLoader());
       if (node instanceof BeamIOSinkRel) {
         return count(options, node);
+      } else if (isLimitQuery(node)) {
+        return limitCollect(options, node);
       }
+
       return collect(options, node);
     } finally {
       Thread.currentThread().setContextClassLoader(originalClassLoader);
     }
   }
 
+  enum LimitState {
+    REACHED,
+    NOT_REACHED
+  }
+
+  private static class LimitStateWrapper implements Serializable {
```

Comment:
Why is there a wrapper? Is this to make it clearly something like an `LVar`?

Reply:
Rename it to LimitStateVar